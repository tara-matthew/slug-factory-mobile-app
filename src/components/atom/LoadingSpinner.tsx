import React from "react";
import { View } from "react-native";
import { ActivityIndicator } from "react-native-paper";

const LoadingSpinner = () => {
    return (
        <View
            className="absolute top-0 left-0 right-0 bottom-0 justify-center bg-white z-10"
        >
            <ActivityIndicator size="large" color="#0000ff" animating={ true } />
        </View>
    );
};

export default LoadingSpinner;
