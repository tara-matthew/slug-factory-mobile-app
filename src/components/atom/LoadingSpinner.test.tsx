import React from "react";
import { render } from "@testing-library/react-native";
import LoadingSpinner from "./LoadingSpinner";

jest.useFakeTimers();

describe("LoadingSpinner", () => {
    afterEach(() => {
        jest.clearAllTimers();
    });

    it("renders correctly", () => {
        const { toJSON } = render(<LoadingSpinner />);
        expect(toJSON()).toMatchSnapshot();
    });

    it("renders an ActivityIndicator", async () => {
        const { getByRole } = render(<LoadingSpinner />);

        const spinner = getByRole("progressbar");
        expect(spinner).toBeTruthy();
    });
});
