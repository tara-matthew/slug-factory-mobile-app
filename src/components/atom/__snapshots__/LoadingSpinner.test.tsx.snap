// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner renders correctly 1`] = `
<View
  className="absolute top-0 left-0 right-0 bottom-0 justify-center bg-white z-10"
>
  <View
    accessibilityRole="progressbar"
    accessibilityState={
      {
        "busy": true,
      }
    }
    accessible={true}
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "height": 48,
          "opacity": 1,
          "width": 48,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "height": 48,
              "transform": [
                {
                  "rotate": "45deg",
                },
              ],
              "width": 48,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "height": 24,
                "overflow": "hidden",
                "width": 48,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "height": 48,
                  "transform": [
                    {
                      "translateY": 0,
                    },
                    {
                      "rotate": "-165deg",
                    },
                  ],
                  "width": 48,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 24,
                    "overflow": "hidden",
                    "width": 48,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "borderColor": "#0000ff",
                      "borderRadius": 24,
                      "borderWidth": 4.8,
                      "height": 48,
                      "width": 48,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "height": 48,
              "transform": [
                {
                  "rotate": "45deg",
                },
              ],
              "width": 48,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "height": 24,
                "overflow": "hidden",
                "top": 24,
                "width": 48,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "height": 48,
                  "transform": [
                    {
                      "translateY": -24,
                    },
                    {
                      "rotate": "345deg",
                    },
                  ],
                  "width": 48,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 24,
                    "overflow": "hidden",
                    "width": 48,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "borderColor": "#0000ff",
                      "borderRadius": 24,
                      "borderWidth": 4.8,
                      "height": 48,
                      "width": 48,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
